### PR TITLE
mandatory calico network CIDR

### PR DESCRIPTION
**Pull request by mainred**
## High-level description

This PR allows the `calico_network_cidr` to be set by the operators, as a mandatory configuration, if windows-support is enabled. Overlapped subnets of calico and infrastructure may bring unrecoverable/tremendous problems into the infrastructure especially for an on-premise environment.

This PR is blocked by the task described in [DCOS-62550](https://jira.mesosphere.com/browse/DCOS-62550), which make calico configurations configurable through terraform.
Also, before we can turn all tests into green, we need to first add`calico_network_cidr` into `main.tf` defined in teamcity project.

## Corresponding DC/OS tickets (required)

  - [DCOS-60734](https://jira.mesosphere.com/browse/DCOS-60734) make calico-network-cidr mandatory to set by the operator

## Related tickets (optional)

This PR is currently blocked my this task
  - [D2IQ-64346](https://jira.d2iq.com/browse/D2IQ-64346) add calico network for cluster running CI.

## Checklist for component/package updates:

If you are changing components or packages in DC/OS (e.g. you are bumping the sha or ref of anything underneath `packages`), then in addition to the above please also include:

  - [ ] Change log from the last version integrated (this should be a link to commits for easy verification and review): [example](https://github.com/dcos/dcos-mesos-modules/compare/f6fa27d7c40f4207ba3bb2274e2cfe79b62a395a...6660b90fbbf69a15ef46d0184e36755881d6a5ae)
  - [ ] Included a test which will fail if code is reverted but test is not. If there is no test please explain.
  - [ ] Test Results: [link to CI job test results for component]
  - [ ] Code Coverage (if available): [link to code coverage report]

### CODE REVIEW

File: gen/calc.py [hunk 1]
```diff
@@ -335,11 +335,17 @@ def validate_config_subnet(config_name, subnet, version=IPVersion.IPv4):
...
             raise ValueError("IP version not match")
     except ValueError as ex:
-        err_msg = "Incorrect value for {}: {}. Only IPv{} subnets are allowed".format(
+        err_msg = "Incorrect value for `{}`: `{}`. Only IPv{} subnets are allowed".format(
             config_name, subnet, version)
         raise AssertionError(err_msg) from ex
 
 
+def validate_calico_network_cidr(calico_network_cidr, enable_windows_agents):
+    if enable_windows_agents.lower() == 'true':
```
Comment (mainred) [inline gen/calc.py]: @jongiddy , I need your confirmation here.
The background is that we do not support calico in windows, so we don't need to validate calico configurations if DC/OS is deployed on windows.
- Can we just verify DC/OS on windows by `enable_windows_agents`?
- Is it possible to upgrade a dc/os cluster to support windows agents from just Linux agents?

Thanks in advance.

Comment (mainred) [reply gen/calc.py]: CC @DominikDary

Comment (DominikDary) [reply gen/calc.py]: 1) yes
2) no

File: gen/build_deploy/bash.py [hunk 2]
```diff
@@ -513,7 +513,8 @@ def validate_custom_check_bins_dir(custom_check_bins_dir):
             "61091 telegraf" \
             "62020 fluent-bit" \
             "62080 dcos-net" \
-            "62501 dcos-net"
+            "62501 dcos-net" \
+            "64000 dcos-net"
```
Comment (mainred) [inline gen/build_deploy/bash.py]: This code was hard-coded in dcos-net for handling VXLAN packets

File: gen/__init__.py [hunk 3]
```diff
@@ -491,6 +491,7 @@ def get_dcosconfig_source_target_and_templates(
         'dcos_image_commit',
         'package_ids',
         'template_filenames',
+        'enable_windows_agents',
```
Comment (mainred) [inline gen/__init__.py]: targets are collected from the variables used in templates, plus a list of variables used outside of the templates.
I added `enable_windows_agents` as a base target, to validate calico_network_cidr when windows is not enabled.

**Comment by mainred** [comment]
Tests failed is expected, for we don't provide calico_network_cidr in `main.tf` used to launch a cluster.

**Comment by mainred** [comment]
@mesosphere-mergebot label Ready For Review

**Comment by mainred** [comment]
@mesosphere-mergebot changelog-not-required for part of calico work

**Comment by mainred** [comment]
@mesosphere-mergebot request-review  @vespian @jkoelker @wavesoft

**Comment by mainred** [comment]
Could anybody please take over this PR? @orlandohohmeier @DominikDary 
It's almost there if we turn dcos-enterpse bump PR green, and I have no access to dcos-enterprise repo and the Teamcity project for further debugging CI issues.

**Comment by orlandohohmeier** [comment]
@mainred we'll take over, again thanks a lot for your contributions and hard work! 🙇

**Comment by wavesoft** [comment]
Opened replacement here : https://github.com/dcos/dcos/pull/7056

**Pull request closed without merging**